lib/engine/part/path.rb

```diff
@@ -51,10 +50,31 @@ def <=>(other)
       end
 
       def <=(other)
+        other_ends = other.ends
+        ends.all? { |t| other_ends.any? { |o| t <= o } }
+      end
+
+      def match?(other)
         (@a <= other.a && @b <= other.b) ||
           (@a <= other.b && @b <= other.a)
       end
 
+      def ends
+        @ends ||= calculate_ends
+      end
+
+      def calculate_ends
+        [@a, @b].flat_map do |j|
+          next [j] unless j.junction?
+
+          j.paths.map do |p|
```

Comment:
this looks like it's gonna break, i wonder why it didn't?

it should be paths.flat_map

next [] if p == self

Reply:
Because it should only affect Lawson tiles, and I hadn't gotten around to retesting them after these changes. My bad.

Reply:
Scratch that. It doesn't break. a and b don't have paths unless they are junctions.

Reply:
Still need to test with lawson tiles

Reply:
The select needs to be a reject